src/lib/y2network/config.rb

```diff
@@ -69,5 +67,22 @@ def initialize(id: :system, interfaces:, routing:, source:)
     def write
       Y2Network::ConfigWriter.for(source).write(self)
     end
+
+    # Returns a deep-copy of the configuration
+    #
+    # @return [Config]
+    def copy
+      Marshal.load(Marshal.dump(self))
+    end
+
+    # Determines whether two configurations are equal
+    #
+    # @return [Boolean] true if both configurations are equal; false otherwise
+    def ==(other)
+      source == other.source && (interfaces - other.interfaces).empty? &&
```

Comment:
is `interfaces - other.interfaces` correct? it can lead to situation that `==` is order dependent, which is very unexpected. I would rather use `interfaces.sort == other.interfaces.sort` or maybe even with uniq if you want to be sure that duplicates does not break it.